### PR TITLE
fix: remove reject_non_steady flag

### PR DESCRIPTION
**Pull request by carrascomj**
The `reject_non_steady` flag is not used anymore in the model, so it should be removed. This is important mainly because there was a regression on the validation of the config, which this PR removes altogether.

Maybe we should discuss what to do with the "penalize" flag.

Checklist:

- [x] Updated any relevant documentation
- [ ] Add an adr doc if appropriate
- [ ] Include links to any relevant issues in the description
- [x] Unit tests passing
- [ ] Integration tests passing

### CODE REVIEW

**Comment by carrascomj** [comment]
This closes #435

File: maud/stan/functions.stan [hunk 1]
```diff
@@ -822,7 +822,7 @@ vector maud_ae_system(vector conc,
                       array[,] int phosphorylation_ix_bounds,
                       array[] int phosphorylation_type,
                       array[] int phosphorylation_pme){
-  return conc - to_vector(ode_bdf_tol(dbalanced_dt, conc, 0, {1},
+  return conc - to_vector(ode_bdf_tol(dbalanced_dt, conc, 0, {1e-6},
```
Comment (teddygroves) [inline maud/stan/functions.stan:825]: I think this line isn't meant to be part of this pr?

**Review by teddygroves — CHANGES_REQUESTED** [review]
Looks good, just one question.

File: maud/stan/functions.stan (hunk 1 above)
Comment (carrascomj) [reply maud/stan/functions.stan]: Ye, sorry.

**Pull request merged**